plugins/inputs/jenkins/jenkins.go

```diff
@@ -432,8 +448,10 @@ func (jr jobRequest) parentsString() string {
 	return strings.Join(jr.parents, "/")
 }
 
-func gatherJobBuild(jr jobRequest, b *buildResponse, acc telegraf.Accumulator) {
-	tags := map[string]string{"name": jr.name, "parents": jr.parentsString(), "result": b.Result}
+func gatherJobBuild(tags map[string]string, jr jobRequest, b *buildResponse, acc telegraf.Accumulator) {
+	tags["name"] = jr.name
+	tags["parents"] = jr.parentsString()
+	tags["result"] = b.Result
```

Comment:
I like the idea but I think this introduces a race condition with this tag map, since the jobs are collected concurrently and there is only one map being written here.

I'd suggest going about this a slightly different way.  If you define a `func (j *Jenkins) Init() error` function it will be called when the plugin is created, you can parse the URL and add the `source` and `port` tags to the Jenkins struct.  Then we can add them to the new map here and we won't need to pass the map down.

Reply:
@danielnelson Code is updated as per the suggestion. Now
> Source and port tags are part of Jenkins struct
> Tags are not passed down to the functions as argument.